Data/ByteString/Builder/Internal.hs

```diff
@@ -414,6 +417,12 @@ instance Monoid Builder where
   {-# INLINE mconcat #-}
   mconcat = foldr mappend mempty
 
+instance IsList Builder where
+  type Item Builder = Word8
+  fromList = byteStringInsert . fromList
```

Comment:
`fromList` ought to be productive for infinite lists as well as finite lists.

Reply:
I have just changed it to use lazyByteStringInsert; is this correct?

Reply:
Yes and no. It still leads to degenerate chunk sizes in uses like `stimes 1000 (fromList [10])`. Just `lazyByteString . fromList` should be OK, I think.